src/pooling.jl

```diff
@@ -201,8 +201,9 @@ For all elements `x` in a size `k` window, lpnormpool computes `(∑ᵢ xᵢ^p)^
 
 Thus `lpnormpool(x, 1, k) ./ prod(k) ≈ meanpool(x, k)` and `lpnormpool(x, 2, k).^2 ./ prod(k) ≈ meanpool(x.^2, k)`.
 """
-function lpnormpool(x, p::Number, k::NTuple{N, Integer}; pad=0, stride=k) where N
+function lpnormpool(x, p::T, k::NTuple{N, Integer}; pad=0, stride=k) where {N,T<:Number}
     (isinf(p) || p < 0) && error("p value of Lp norm pool expects `0 < p < Inf`, but p is $(p) now.")
+    (typeof(p) <: Integer) || (p = convert(eltype(x), p)) # p needn't convert only if typeof(p) <: Integer
     pad = expand(Val(N), pad)
     stride = expand(Val(N), stride)
     pdims = PoolDims(x, k; padding=pad, stride=stride)
```

Comment:
I suggest never re-assigning something of a different type to the same variable name, and especially not within an `if` statement. Sometimes this causes type instability, sometimes it confuses Zygote, often it confuses readers.

I also think only real `p` is accepted. 

And that comments repeating the code aren't necessary.

```suggestion
function lpnormpool(x, pow::Real, k::NTuple{N, Integer}; pad=0, stride=k) where {N}
    p = pow isa Integer ? pow : convert(float(eltype(x)), pow)
    (isinf(p) || p < 0) && error("p value of Lp norm pool expects `0 < p < Inf`, but p is $(p) now.")
    pdims = PoolDims(x, k; padding=expand(Val(N), pad), stride=expand(Val(N), stride))
```

Reply:
OK! I fix this.